### DISCUSSION TITLE
Adapting Librispeech recipes to use a pronunciation dictionary

### DISCUSSION THREAD

**Discussion by maltium** [k2-fsa/icefall#788] [General]
Most of the work seems to be going into the Librispeech recipes and although the preparation script creates _lang_phone_, it seems that all recipes work using character bpe.

Is it possible to make Librispeech recipes work using the traditional pronunciation dictionary? Maybe past experiments showed that this yields inferior results?

**Comment by csukuangfj** [comment]
We have a recipe using phone-based pronunciation. Please have a look at the tdnn_lstm_ctc folder.

Be aware that the above folder is quite old but it provides an example about how to use phone-based instead of BPE-based modeling units.

**Reply by maltium** [reply]
Any insight while phone-based modeling units approach is mostly abandoned?

**Reply by csukuangfj** [reply]
> Any insight while phone-based modeling units approach is mostly abandoned?

BPE-based modeling usually yields lower WER. Also, it does not need a pronunciation dictionary and the BPE model can be trained from the data.

Phone-based modeling is still useful. Its vocabulary size is smaller and the resulting search graph and search space are also smaller.
